### PR TITLE
Cdd 1913 - move call to metrics to just before push to customs notifications gateway

### PR DESCRIPTION
**Pull request by mattclark-zerogravit**

### CODE REVIEW

File: app/uk/gov/hmrc/customs/notification/services/PushClientNotificationService.scala [hunk 1]
```diff
@@ -32,14 +33,21 @@ import scala.concurrent.duration.Duration
...
 
   private implicit val hc = HeaderCarrier()
 
   def send(declarantCallbackData: DeclarantCallbackData, clientNotification: ClientNotification): Boolean = {
 
     val pushNotificationRequest = pushNotificationRequestFrom(declarantCallbackData, clientNotification)
 
+    clientNotification.metricsStartDateTime.fold() { startTime =>
+      metricsConnector.post(CustomsNotificationsMetricsRequest(
+        "NOTIFICATION", clientNotification.notification.conversationId,DateTimeUtils.convertDateTimeToZonedDateTime(startTime), dateTimeService.zonedDateTimeUtc))
```
Comment (googley42) [inline app/uk/gov/hmrc/customs/notification/services/PushClientNotificationService.scala]: `,DateTimeUtils` should be `, DateTimeUtils`

File: test/acceptance/PushPullNotificationsSpec.scala [hunk 2]
```diff
@@ -111,7 +112,7 @@ class PushPullNotificationsSpec extends AcceptanceTestSpec
 
     notificationRepo.save(
       ClientNotification(ClientSubscriptionId(expectedCall.client.csid),
-        Notification(ConversationId(expectedCall.conversationId), headers, expectedCall.xml.toString(), MimeTypes.XML)))
+        Notification(ConversationId(expectedCall.conversationId), headers, expectedCall.xml.toString(), MimeTypes.XML), Some(TimeReceivedDateTime), None))
```
Comment (googley42) [inline test/acceptance/PushPullNotificationsSpec.scala]: `None` parameter should use param name

File: test/unit/services/PushClientNotificationServiceSpec.scala [hunk 3]
```diff
@@ -37,24 +38,41 @@ class PushClientNotificationServiceSpec extends UnitSpec with MockitoSugar with
   private val mockPushNotificationServiceConnector = mock[PushNotificationServiceConnector]
   private val mockGAConnector = mock[GoogleAnalyticsSenderConnector]
   private val mockNotificationLogger = mock[NotificationLogger]
```
Comment (googley42) [inline test/unit/services/PushClientNotificationServiceSpec.scala]: `private val mockCustomsNotificationsMetricsConnector = mock[CustomsNotificationMetricsConnector]` is not used

File: test/unit/controllers/CustomsNotificationControllerSpec.scala [hunk 4]
```diff
@@ -91,11 +88,10 @@ class CustomsNotificationControllerSpec extends UnitSpec with Matchers with Mock
 
     "respond with status 202 for valid request" in {
       returnMockedCallbackDetailsForTheClientIdInRequest()
-      when(mockCustomsNotificationsMetricsConnector.post(any[CustomsNotificationsMetricsRequest])).thenReturn(Future.successful(()))
+
       testSubmitResult(ValidRequestWithMixedCaseCorrelationId) { result =>
         status(result) shouldBe ACCEPTED
       }
-        verifyMetricsConnector
```
Comment (googley42) [inline test/unit/controllers/CustomsNotificationControllerSpec.scala]: spacing

File: app/uk/gov/hmrc/customs/notification/services/PushClientNotificationService.scala (hunk 1 above)
Comment (mattclark-zerogravit) [reply app/uk/gov/hmrc/customs/notification/services/PushClientNotificationService.scala]: ok will fix

File: test/acceptance/PushPullNotificationsSpec.scala (hunk 2 above)
Comment (mattclark-zerogravit) [reply test/acceptance/PushPullNotificationsSpec.scala]: OK will fix

File: test/unit/services/PushClientNotificationServiceSpec.scala (hunk 3 above)
Comment (mattclark-zerogravit) [reply test/unit/services/PushClientNotificationServiceSpec.scala]: This is used as interactions with this are verified in the tests

File: test/unit/controllers/CustomsNotificationControllerSpec.scala (hunk 4 above)
Comment (mattclark-zerogravit) [reply test/unit/controllers/CustomsNotificationControllerSpec.scala]: ok will fix

**Pull request merged**